### DISCUSSION TITLE
Can apisix not depend on etcd?

### DISCUSSION THREAD

**Discussion by zxyao145** [apache/apisix#10693] [Q&A]
Is it possible to store configuration information such as plugins elsewhere, such as nacos, mysql, etc.

**Comment by zxyao145** [comment]
@spacewander Hi, could you possibly assist me by answering this question?

**Reply by spacewander** [reply]
I don't work on this project anymore, so my knowledge is not up-to-date. You can store the configuration in a static file, see https://github.com/apache/apisix/blob/master/docs/en/latest/deployment-modes.md#standalone.

**Comment by zxyao145** [comment]
> I don't work on this project anymore, so my knowledge is not up-to-date. You can store the configuration in a static file, see https://github.com/apache/apisix/blob/master/docs/en/latest/deployment-modes.md#standalone.

I'm very sorry to bother you. And thank you for your answer!

**Comment by flylan** [comment]
You can save the configuration in MySQL or SQLite, but you need to use an ETCD converter, such as Kine.https://github.com/k3s-io/kine

However, this requires a certain level of technical expertise to transform, as Apisix is generally strongly coupled with ETCD

Although Apisix can store configurations in yaml without relying on ETCD, once you choose to store configurations in yaml, you cannot use the API to update configurations.

**Reply by zxyao145** [reply]
Thank you very much for your answer. I think Kine is a better answer.

**Reply by flylan** [reply]
I am also looking for alternative solutions to ETCD. As far as I know, Apisix relies on the characteristics of ETCD in its underlying design, which makes it difficult to replace it easily. Currently, Apisix only supports saving configurations to ETCD or YAML files. In the absence of official support for other storage, conversion can only be done through Kine for the time being.

**Comment by flylan** [comment]
Etcd is a relatively new service that was originally born as a core component of k8s. Unfortunately, there is a lack of maintainers currently available, and most operations are not very familiar with maintaining this service. I estimate that most operations are more familiar with MySQL. Refer to this explanation: https://blog.frankel.ch/apisix-without-etcd/

When the official does not yet support other storage services, Kine may be the best choice. Moreover, Kine serves as a K3s service dependency component, simulating some of the interfaces of ETCD to replace ETCD services. However, it should be noted that Kine must run as a resident service, as it simulates the interface of ETCD. This will increase the complexity of the system.

**Reply by zxyao145** [reply]
You are so kind! This post has greatly inspired me.
